### ISSUE TITLE
variance very tight[Bug]

### ISSUE THREAD

**Issue by galfaroi** [cornellius-gp/gpytorch#1227]
# 🐛 Bug
I think this is not a bug... the variance I getting is very tight almost the same as the mean, when is obviously not the case. I am using grid inducing points regression, I wonder if there is any rule of thumb or any resource that help to overcome this limitation. 

## My code is based on the grid inducing point example

** Code snippet to reproduce **
```python
# Your code goes here
# Please make sure it does not require any external dependencies (other than PyTorch!)
# (We much prefer small snippets rather than links to existing libraries!)
```

** Stack trace/error message **
```
// Paste the bad output here!
```

## System information

**Please complete the following information:**
- 
- 
-

**Comment by jacobrgardner** [comment]
Hello,

You're probably gonna need to provide more information like code or a plot or something similar for us to help with this.

**Comment by galfaroi** [comment]
`
    class GPRegressionModel(gpytorch.models.ExactGP):
        def __init__(self, train_x, train_y, likelihood):
            super(GPRegressionModel, self).__init__(train_x, train_y, likelihood)
         
            self.mean_module = gpytorch.means.ConstantMean()
            self.covar_module = gpytorch.kernels.GridInterpolationKernel( 
              gpytorch.kernels.ScaleKernel(gpytorch.kernels.RQKernel()),
              num_dims=data_dim, grid_size = 15)             
                                                    
        def forward(self, x):
           
            mean_x = self.mean_module(x)
            covar_x = self.covar_module(x).add_jitter(2e-3)
            
            return gpytorch.distributions.MultivariateNormal(mean_x, covar_x)
"""
usual optimization loop and the:
"""
# Get into evaluation (predictive posterior) mode
    model.eval()
    likelihood.eval()
    #model.named_parameters
    #test_x = torch.tensor(test_x, dtype=torch.float32)
    #test_x.dtype
    f_preds = model(test_x)
    y_preds = likelihood(model(test_x))

    f_mean = f_preds.mean
    f_var = f_preds.variance
    f_mean = f_mean.cpu() 
    f_var = f_var.cpu()
    f_mean_numpy = f_mean.detach().numpy()
    f_mean_numpy = np.round(f_mean_numpy)
    #f_mean_numpy
    f_var_numpy = f_var.detach().numpy()

`
some of the outputs:
mean : 123.,       444.,    486.,      289.,           83.,      117.,     637.,      293.,     165 ...
upper: 123.02,  444.01,  486.09,  289.0217,  83.26,  117.02,  637.05,  293.02, 165.02 ...
lower:  122.97  443.98,  485.90,  288.97,       82.73,  116.97,  636.94,  292.97, 164.97 ....

I hope this helps, thanks in advance

**Comment by gpleiss** [comment]
@galfaroi what does your data look like?

**Comment by galfaroi** [comment]
Unfortunately I cannot share data, but is multivariate time series of supply-chain, with one output. GPs are giving better results than RF, but the problem with the data is that is in the order of thousands (rows), so a exact GP  doesn't fit in a GPU, and is not enough data for SGP's work well (at least I found that) so using grid interpolation seems to be working, but there is the issue of uncertainty been too tight when is not. I certainly wont consider the data noiseless. Autocorrelation disappear very soon.

**Comment by gpleiss** [comment]
@galfaroi What happens when you wrap your prediction code with the `with gpytorch.settings.fast_pred_var(True):` context manager? e.g.

```python
model.eval()
likelihood.eval()
#model.named_parameters
#test_x = torch.tensor(test_x, dtype=torch.float32)
#test_x.dtype
with gpytorch.settings.fast_pred_var(True):
    f_preds = model(test_x)
    y_preds = likelihood(model(test_x))

f_mean = f_preds.mean
f_var = f_preds.variance
f_mean = f_mean.cpu() 
f_var = f_var.cpu()
f_mean_numpy = f_mean.detach().numpy()
f_mean_numpy = np.round(f_mean_numpy)
#f_mean_numpy
f_var_numpy = f_var.detach().numpy()
```
